presto-main/src/main/java/io/prestosql/operator/LookupJoinOperator.java

```diff
@@ -550,13 +550,14 @@ private void processProbe(LookupSource lookupSource)
                             break;
                         }
                     }
+                    // reset row join state for next row
+                    currentProbePositionProducedRow = false;
+                    statisticsCounter.recordProbe(joinSourcePositions);
+                    joinSourcePositions = 0;
                 }
-                currentProbePositionProducedRow = false;
                 if (!advanceProbePosition(lookupSource)) {
                     break;
                 }
-                statisticsCounter.recordProbe(joinSourcePositions);
```

Comment:
we should move `recordProbe` (as you did) & leave the others here.

What we should do, however, is reset `joinSourcePositions` and `currentProbePositionProducedRow` in the same place (currently not the case). Maybe let's move them all to `advanceProbePosition`, after `joinPosition = probe.getCurrentJoinPosition(lookupSource);` line

Reply:
> What we should do, however, is reset joinSourcePositions and 

good idea.